app/scripts/background.js

```diff
@@ -678,6 +678,7 @@ export function setupController(
   //
 
   updateBadge();
+  controller.txController.addInitialTransactionApprovals(updateBadge);
```

Comment:
If we end up needing to restore the pending approvals for multiple controllers, we might want to come up with a more generic name so that we can call the same method by just passing a list of controllers here.

But that's only if that ends up being the case. Also, we can change it afterwards if needed, so this is just an observation.

Reply:
I'd suggested the name `initApprovals` which would be generic enough to support this.